openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -153,23 +160,170 @@ def absolute_url(relative_path):
         yield (user, first_schedule.enrollment.course.language, template_context)
 
 
-def encode_url(url):
-    # Sailthru has a bug where URLs that contain "+" characters in their path components are misinterpreted
-    # when GA instrumentation is enabled. We need to percent-encode the path segments of all URLs that are
-    # injected into our templates to work around this issue.
-    parsed_url = urlparse(url)
-    modified_url = parsed_url._replace(path=urlquote(parsed_url.path))
-    return modified_url.geturl()
+@task(ignore_result=True, routing_key=ROUTING_KEY)
+def recurring_nudge_schedule_bin(
+    site_id, target_day_str, day_offset, bin_num, org_list, exclude_orgs=False, override_recipient_email=None,
+):
+    target_day = deserialize(target_day_str)
+    msg_type = RecurringNudge(abs(day_offset))
+
+    for (user, language, context) in _recurring_nudge_schedules_for_bin(target_day, bin_num, org_list, exclude_orgs):
+        msg = msg_type.personalize(
+            Recipient(
+                user.username,
+                override_recipient_email or user.email,
+            ),
+            language,
+            context,
+        )
+        _recurring_nudge_schedule_send.apply_async((site_id, str(msg)), retry=False)
+
+
+def _recurring_nudge_schedules_for_bin(target_day, bin_num, org_list, exclude_orgs=False):
+    beginning_of_day = target_day.replace(hour=0, minute=0, second=0)
+    users = User.objects.filter(
+        courseenrollment__schedule__start__gte=beginning_of_day,
+        courseenrollment__schedule__start__lt=beginning_of_day + datetime.timedelta(days=1),
+        courseenrollment__is_active=True,
+    ).annotate(
+        first_schedule=Min('courseenrollment__schedule__start')
```

Comment:
we don't need this any more, right?

Reply:
I removed both of these annotations when refactoring the query into separate functions. It looks like we just pick the first schedule in the list regardless of this annotation.